### PR TITLE
OCI: Add kafka unit tests

### PR DESCRIPTION
**Pull request by athos-ribeiro**
Signed-off-by: Athos Ribeiro <athos.ribeiro@canonical.com>

### CODE REVIEW

File: oci-unit-tests/kafka_test.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+# Helper function to execute zookeeper with some common arguments.
+docker_run_zookeeper() {
+    docker run \
+     --network "$DOCKER_NETWORK" \
+     --rm \
+     -d \
+     -p "${zookeeper_port}":"${zookeeper_port}" \
+     --name zookeeper_test_${id} \
```
Comment (sergiodj) [inline oci-unit-tests/kafka_test.sh:60]: I think that if you prefix the container name with `${DOCKER_PREFIX}` then you won't need to loop twice in the `tearDown` function.

File: oci-unit-tests/kafka_test.sh [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      --topic quickstart-events --bootstrap-server \
+      kafka_test_${id}:${broker_port} < "${tempdir}/kafka_oci_test_message"
+
+    debug "Consuming test message"
+    docker exec "${client_container}" kafka-console-consumer.sh \
+      --max-messages 1 --timeout-ms 5000 --topic quickstart-events \
+      --from-beginning --bootstrap-server kafka_test_${id}:${broker_port} \
+      2>&1 | grep -qF 'oci test message'
+    assertTrue "Could not consume message. Was the message produced?" $?
+}
```
Comment (sergiodj) [inline oci-unit-tests/kafka_test.sh:174]: You also have to `rm -rf ${tempdir}` before the function returns.

File: oci-unit-tests/kafka_test.sh [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+# Helper function to execute kafka with some common arguments.
+docker_run_broker() {
+    docker run \
+     --network "$DOCKER_NETWORK" \
+     --rm \
+     -d \
+     -p "${broker_port}":"${broker_port}" \
+     --name kafka_test_${id} \
```
Comment (sergiodj) [inline oci-unit-tests/kafka_test.sh:72]: Same comment about using `${DOCKER_PREFIX}` here.

**Review by sergiodj — APPROVED** [review]
Thanks for the PR, @athos-ribeiro.  Nice set of tests!  I found some minor nits to fix, but otherwise this LGTM so I'm approving it now.

**Comment by athos-ribeiro** [comment]
Thanks for the review, Sergio!

I addressed, all your comments and rebased my branch on top of `main`

**Pull request merged**